src/testrel.jl

```diff
@@ -517,13 +532,33 @@ function _execute_test(
     engine::String,
     program::String,
     timeout_sec::Int64,
-    readonly::Bool,
+    readonly::Bool;
+    retry_number=1,
 )
     @debug("$name: Starting execution")
-    rsp = exec_async(context, schema, engine, program;
-            readtimeout = timeout_sec, readonly)
+    rsp = try
+        # Exec async really should return after 2-3 seconds
+        exec_async(context, schema, engine, program; readtimeout=30, readonly)
+    catch e
```

Comment:
Should we be more specific in this exception (e.g. what if one of the arguments is actually wrong)? What exception normally comes out?

Reply:
We've seen a couple different exceptions here, so being specific didn't seem worth it in this case. (Also required taking on a direct dependency on `HTTP`, which felt meh).

A connection failed is most common, but sometimes just a request exception.

Reply:
The logs do include the original error message and if the test ultimately fails, the full stacktrace.